runtime/os_interface/linux/drm_neo.cpp

```diff
@@ -25,7 +25,7 @@ int Drm::ioctl(unsigned long request, void *arg) {
     int ret;
     SYSTEM_ENTER();
     do {
-        ret = ::ioctl(fd, request, arg);
+        ret = ::ioctl(fd, static_cast<unsigned int>(request), arg);
```

Comment:
Why do you need this cast?

Reply:
Android NDK compiler says cannot convert from unsigned long to unsigned int.

Reply:
Linux `ioctl` signature is:

> int ioctl(int fd, unsigned long request, ...);

from here: http://man7.org/linux/man-pages/man2/ioctl.2.html

Android's declaration looks like:

> int ioctl(int __fd, int __request, ...);

https://android.googlesource.com/platform/bionic/+/master/libc/include/bits/ioctl.h#43

---

On x86 Linux-based systems both `int` and `long` are 32-bit.

Perhaps `#if defined(__ANDROID__)` guard can help here.

Reply:
Thanks! Replaced.

Reply:
can you perform the separation at cmake level? I mean one file for linux, one file for android - cmake chooses correct file.